components/camel-jira/src/main/java/org/apache/camel/component/jira/consumer/AbstractJiraConsumer.java

```diff
@@ -79,18 +79,28 @@ protected List<Issue> getIssues() {
     protected List<Issue> getIssues(String jql, int start, int maxPerQuery, int maxResults) {
         LOG.debug("Start indexing current JIRA issues...");
 
+        if (maxResults < maxPerQuery) {
+            maxPerQuery = maxResults;
+        }
+
         List<Issue> issues = new ArrayList<>();
         while (true) {
             SearchRestClient searchRestClient = endpoint.getClient().getSearchClient();
-            SearchResult searchResult = searchRestClient.searchJql(jql, maxResults, start, null).claim();
+            SearchResult searchResult = searchRestClient.searchJql(jql, maxPerQuery, start, null).claim();
 
             for (Issue issue : searchResult.getIssues()) {
-                issues.add(issue);
+                // Avoid duplicates
+                if (!issues.contains(issue)) {
+                    issues.add(issue);
```

Comment:
What do you think of leaving it like before and simply replacing the `ArrayList` with a `LinkedHashSet` to avoid calling `contains` on a list?

Reply:
@essobedo thanks

Reply:
@oscerd could you please (re)approve the workflow as I force pushed the review changes